src/zarr/core/codec_pipeline.py

```diff
@@ -320,6 +309,20 @@ def _merge_chunk_array(
                     for idx in range(chunk_spec.ndim)
                 )
                 chunk_value = chunk_value[item]
+        if is_complete_chunk and chunk_value.shape == chunk_spec.shape:
```

Comment:
this will fail for scalars, need a test for that too.

Reply:
Well it works for scalars, because `chunk_value` is now NDBuffer which does have `shape`. I added a test.